rust/rope/src/engine.rs

```diff
@@ -482,15 +471,222 @@ impl Engine {
         }
         self.revs.reverse();
     }
+
+    /// Merge the new content from another Engine into this one with a CRDT merge
+    pub fn merge(&mut self, other: &Engine) {
+        let (mut new_revs, text, tombstones, deletes_from_union) = {
+            let base_index = find_base_index(&self.revs[..], &other.revs[..]);
```

Comment:
I wasn't sure if it would work without them, and I guess in the interest of avoiding long Rust compile loops, I just went with them. Fixed.